src/bin/tikv-ctl.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![feature(plugin)]
+#![cfg_attr(feature = "dev", plugin(clippy))]
+
+extern crate tikv;
+extern crate clap;
+extern crate protobuf;
+extern crate kvproto;
+extern crate rocksdb;
+
+use std::{str, u64};
+use clap::{Arg, App, SubCommand};
+use protobuf::Message;
+use kvproto::raft_cmdpb::RaftCmdRequest;
+use kvproto::raft_serverpb::{RaftLocalState, RegionLocalState, RaftApplyState};
+use kvproto::eraftpb::Entry;
+use rocksdb::DB;
+use tikv::util::{self, escape, unescape};
+use tikv::raftstore::store::keys;
+use tikv::raftstore::store::engine::{Peekable, Iterable};
+use tikv::storage::ALL_CFS;
+use tikv::storage::CF_RAFT;
+
+static VERSION: &'static str = "0.0.1";
+
+fn main() {
+    let matches = App::new("tikv")
```

Comment:
Actually, `util::escape`, `util::unescape` is used in this tool. There is nothing related to encoding/decoding. I think tests are not needed here.